gcloud-java-logging/README.md

```diff
@@ -15,32 +17,155 @@ Java idiomatic client for [Google Cloud Logging][cloud-logging].
+credentials. You can then make API calls by calling methods on the Logging service object. The
+simplest way to authenticate is to use
+[Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
+These credentials are automatically inferred from your environment, so you only need the following
+code to create your service object:
+
+```java
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.LoggingOptions;
+
+LoggingOptions options = LoggingOptions.defaultInstance();
+try(Logging logging = options.service()) {
+  // use logging here
+}
+```
+
+For other authentication options, see the
+[Authentication](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) page.
+
+#### Creating a metric
+With Logging you can create logs-based metrics. Logs-based metrics allow to keep track of the number
+of log messages associated to specific events. Add the following imports at the top of your file:
+
+```java
+import com.google.cloud.logging.Metric;
+import com.google.cloud.logging.MetricInfo;
+```
+Then, to create the metric, use the following code:
+
+```java
+MetricInfo metricInfo = MetricInfo.builder("test-metric", "severity >= ERROR")
+    .description("Log entries with severity higher or equal to ERROR")
+    .build();
+logging.create(metricInfo);
+```
+
+#### Writing log entries
+With Logging you can also write custom log entries. Add the following imports at the top of your
+file:
+```java
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Payload.StringPayload;
 
-TODO: link to docs on activating Logging, high-level documentation on
-the API, and code snippet
+import java.util.Collections;
+```
+Then, to write the log entries, use the following code:
+```java
+LogEntry firstEntry = LogEntry.builder(StringPayload.of("message"))
+    .logName("test-log")
+    .resource(MonitoredResource.builder("global")
+        .addLabel("project_id", options.projectId())
+        .build())
+    .build();
+logging.write(Collections.singleton(firstEntry));
+```
+
+#### Add a Stackdriver Logging handler to a logger
+You can also register a `LoggingHandler` to a `java.util.logging.Logger` that publishes log entries
+to Stackdriver Logging. Given the following logger:
+```java
+private final static Logger LOGGER = Logger.getLogger(MyClass.class.getName());
+```
+You can register a `LoggingHandler` with the code:
+```java
+LoggingHandler.addHandler(LOGGER, new LoggingHandler());
+```
+After that, logs generated using `LOGGER` will be also directed to Stackdriver Logging.
+
+Notice that you can also register a `LoggingHandler` via the `logging.properties` configuration
+file. Adding, for instance, the following line:
+```
+com.google.cloud.examples.logging.snippets.AddLoggingHandler.handlers=com.google.cloud.logging.LoggingHandler
+```
```

Comment:
There could be depending on your application needs. But I would say that this does not depend on our handler but rather it's a design choice that comes with JUL.